internal/node/node.bzl

```diff
@@ -247,23 +254,27 @@ def _nodejs_binary_impl(ctx):
         is_executable = True,
     )
 
-    runfiles = depset(node_tool_files + ctx.files._bash_runfile_helpers + [ctx.outputs.loader, ctx.file._repository_args], transitive = [sources, node_modules])
+    runfiles_depsets = []
+    runfiles_depsets.extend(node_tool_files)
+    runfiles_depsets.extend(ctx.files._bash_runfile_helpers)
+    runfiles_depsets.append(ctx.outputs.loader)
+    runfiles_depsets.append(ctx.file._repository_args)
 
     if is_windows(ctx):
-        runfiles = depset([ctx.outputs.script], transitive = [runfiles])
+        runfiles_depsets.append(ctx.outputs.script)
         executable = create_windows_native_launcher_script(ctx, ctx.outputs.script)
     else:
         executable = ctx.outputs.script
 
     # entry point is only needed in runfiles if it is a .js file
     if ctx.file.entry_point.extension == "js":
-        runfiles = depset([ctx.file.entry_point], transitive = [runfiles])
+        runfiles_depsets.append(ctx.file.entry_point)
 
     return [
         DefaultInfo(
             executable = executable,
             runfiles = ctx.runfiles(
-                transitive_files = runfiles,
+                transitive_files = depset(runfiles_depsets),
```

Comment:
the fact that you don't use `transitive = runfiles_depsets` makes me think the variable is misnamed, and runfiles_depsets is actually just runfiles list

Reply:
Yeah I guess it is just plain files, previously those plain files were converted into transitive `depset`s though. I think this change is ok though, but the var should be renamed...

Reply:
Updated to just `runfiles`